### PR TITLE
Fb 1.8.0 snapshot

### PR DESCRIPTION
**Pull request by labkey-susanh**

### CODE REVIEW

**Review by ilikepies101 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -9,7 +9,19 @@ but also make certain assumptions that you may not want to impose on your module
 
 ## Release Notes
 
-### veresion 1.7.0
+### version ???
```
Comment (ilikepies101) [inline README.md]: Should we change this version number?

**Review by ilikepies101 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (labkey-susanh) [reply README.md]: This seems a reasonably clear version number to me. :-) (Yes, thanks for the reminder.)

**Comment by labkey-tchad** [comment]
Verifying on TeamCity:
Build: https://teamcity.labkey.org/viewLog.html?buildId=835460
BVTs: https://teamcity.labkey.org/viewLog.html?buildId=835442
Installers: https://teamcity.labkey.org/viewLog.html?buildId=835625

**Comment by labkey-tchad** [comment]
Rerunning installer after build fix: https://teamcity.labkey.org/viewLog.html?buildId=835921
Verifying alternate JAVA_HOME for tomcat: https://teamcity.labkey.org/viewLog.html?buildId=835980

File: src/main/groovy/org/labkey/gradle/plugin/FileModule.groovy [hunk 2]
```diff
@@ -405,38 +407,69 @@ class FileModule implements Plugin<Project>
...
+                            apiLib(MavenPublication) { pub ->
+                                pub.groupId = LabKeyExtension.API_GROUP
+                                pub.artifact(project.tasks.jar)
                             }
                         }
                     }
 
                     if (BuildUtils.shouldPublish(project))
                     {
                         project.artifactoryPublish {
```
Comment (labkey-tchad) [inline src/main/groovy/org/labkey/gradle/plugin/FileModule.groovy:452]: Can/should we do lazy task configuration here? Something like:
```suggestion
                         project.tasks.named("artifactoryPublish").configure {
```

**Comment by labkey-tchad** [comment]
Build, BVT, and Installers look good. Custom JAVA_HOME for Tomcat works (The run "failed" because I intentionally picked a JDK that we no longer support).
I've switched the TeamCity trunk project to run 1.8.0-SNAPSHOT by default to get more thorough coverage and make sure I'm not missing some corner case.
Any further changes should be verified in a feature branch before adding them to fb_1.8.0-snapshot.

**Review by labkey-tchad — APPROVED** [review]
Just one possible place to use lazy task configuration.

File: src/main/groovy/org/labkey/gradle/plugin/FileModule.groovy (hunk 2 above)
Comment (labkey-susanh) [reply src/main/groovy/org/labkey/gradle/plugin/FileModule.groovy:452]: Probably a good idea, but not needed at the moment.  This is a task that only gets configured when the property is set to do publishing, so not a lot of savings in any case.

**Review by ilikepies101 — APPROVED** [review]

**Pull request merged**